chemreg/compound/views.py

```diff
@@ -80,8 +142,8 @@ class QueryStructureTypeViewSet(DeprecateDeleteMixin, ModelViewSet):
     serializer_class = QueryStructureTypeSerializer
 
 
-class CompoundViewSet(ReadOnlyModelViewSet):
+class CompoundViewSet(SoftDeleteCompoundMixin, ReadOnlyModelViewSet):
```

Comment:
if this is ReadOnly there is no need for a SoftDelete here, but looking at what the mixin provides we may have to split it out into 2 mixins, tough to say, the ticket didn't really talk about this view but I get that it is necessary to moderate responses here after a SoftDelete. Sorry to make this more complicated than it already is, maybe some of the methods in this `SoftDeleteCompoundMixin` can be applied directly to the viewsets in `jsonapi`, for instance having access to the `_is_admin` in any viewset would probably be useful. This seems like such a minor thing but saying exactly how to deal with this problem is pretty difficult, it might make sense to sketch out how each view will deal with a SoftDelete as well as the Deprecation of structures and see if we can find a way to consolidate methods to serve them both, I think it would be worth us all taking a look at this for 20 min. or so and see what we can come up with together.

Reply:
`SoftDeleteCompoundMixin` provides more functionality than just the mechanisms of DELETE. It also provides logic for what is returned from a GET request.

I like having them put together because the functionality is so closely tied... even though the DELETE logic never gets used here. If we ever decide to make `/compounds/` read-write, nothing more needs to be done.

Reply:
> maybe some of the methods in this SoftDeleteCompoundMixin can be applied directly to the viewsets in jsonapi, for instance having access to the _is_admin in any viewset would probably be useful.

I agree that some methods would be useful across many views. Same thing with serializers. We have something of the sorts available on models with the `CommonInfo` class. Maybe we can have a Common `ViewSet` or `Serializer` too. As long as it's easy to find and reason about. These kinds of decisions are fun. Makes me think of this article: [The Wrong Abstraction](https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction).

I personally don't like having them in `jsonapi` though. That package really should just address the ways to go from DRF JSON responses/requests to JSON:API responses/requests. I'd get lost trying to find where `_is_admin` is if it were in there.